dwcj-engine/src/main/java/org/dwcj/component/label/Label.java

```
@@ -3,59 +3,237 @@
+    return this;
+  }
+
+  /**
+   * Alias for the addRightMouseDownListener method.
+   *
+   * @see Label#addRightMouseDownListener(EventListener)
+   * @param listener the event listener to be added
+   * @return The Label itself
+   */
+  public Label onRightMouseDown(EventListener<RightMouseDownEvent> listener) {
+    return addRightMouseDownListener(listener);
+  }
+
+  /**
+   * Removes a RightMouseDown event from the Label component.
+   *
+   * @param listener the event listener to be removed
+   * @return The Label itself
+   */
+  public Label removeRightMouseDownListener(EventListener<RightMouseDownEvent> listener) {
+    dispatcher.removeEventListener(RightMouseDownEvent.class, listener);
+    if (this.ctrl != null && this.dispatcher.getListenersCount(RightMouseDownEvent.class) == 0) {
+      this.rightMouseDownEventSink.removeCallback();
+    }
+    return this;
+  }
+
+  /**
+   * Sets whether the lines will be wrapped in the label.
+   *
+   * @param wrap - Specifies whether the lines will be wrapped (false = Not Wrapped, true = Wrapped)
+   * @return Returns this
+   */
+  public Label setWrap(Boolean wrap) {
+    if (this.ctrl != null) {
+      try {
+        component.setLineWrap(wrap);
+      } catch (BBjException e) {
+        Environment.logError(e);
+      }
+    }
+    this.lineWrap = wrap;
+    return this;
+  }
+
+  /**
+   * Returns whether lines are wrapped in the label.
+   *
+   * @return Returns whether the lines are wrapped in the component (false = Not Wrapped, true =
+   *         Wrapped).
+   */
+  public Boolean isWrap() {
+    return this.lineWrap;
   }
 
+  /**
+   * {@inheritDoc}
+   */
   @Override
   public Label setTextAlignment(Alignment alignment) {
     if (this.ctrl != null) {
       try {
-        ((BBjStaticText) ctrl).setAlignment(alignment.textPosition);
+        component.setAlignment(alignment.textPosition);
       } catch (BBjException e) {
         Environment.logError(e);
```

Don't suppress errors. Throw a DWC run time exception